quickjs.c

```diff
@@ -34135,8 +34140,17 @@ static int JS_WriteObjectAtoms(BCWriterState *s)
 
     bc_put_leb128(s, s->idx_to_atom_count);
     for(i = 0; i < s->idx_to_atom_count; i++) {
-        JSAtomStruct *p = rt->atom_array[s->idx_to_atom[i]];
-        JS_WriteString(s, p);
+        JSAtom atom = s->idx_to_atom[i];
+        if (__JS_AtomIsConst(atom)) {
+            bc_put_u8(s, 0 /* the type */);
+            bc_put_u32(s, atom);
```

Comment:
The encoding for tagged integers could be more efficient, and for keyword-ish atoms (`0 < atom < JS_ATOM_END`) a single byte is enough, for now, because we only have 219.

Not so important however and fine to leave it for another time (but please add a TODO comment in that case.)